File: tests/programs/prog_fibonacci.cpp
#include <iostream>

long long fibonacci(int n) {
    if (n <= 1)
        return n;
    long long a = 0, b = 1;
    for (int i = 2; i <= n; i++) {
        const long long c = a + b;
        a = b;
        b = c;
    }
    return b;
}

int main() {
    constexpr int n = 20;
    for (int i = 0; i <= n; i++)
        std::cout << fibonacci(i) << std::endl;
}

File: tests/programs/prog_fibonacci.rs
fn fibonacci(n: u32) -> u64 {
    match n {
        0 | 1 => n as u64,
        _ => {
            let mut a = 0u64;
            let mut b = 1u64;
            for _ in 2..=n {
                let c = a + b;
                a = b;
                b = c;
            }
            b
        }
    }
}

fn main() {
    const N: u32 = 20;
    (0..=N)
        .map(fibonacci)
        .for_each(|fib| println!("{fib}"));
}
